rct229/rulesets/ashrae9012019/section1/section1rule5.py

```diff
@@ -0,0 +1,203 @@
+            and (pbp_nre_set[0] - pbp_set[0]) / bbp_set[0] > APPLICABLE_LIMIT
+        )
+
+    def get_calc_vals(self, context, data=None):
+        rmd_u = context.USER
+        rmd_b0 = context.BASELINE_0
+        rmd_b90 = context.BASELINE_90
+        rmd_b180 = context.BASELINE_180
+        rmd_b270 = context.BASELINE_270
+        rmd_p = context.PROPOSED
+
+        pbp_set = []
+        bbp_set = []
+        pbp_nre_set = []
+        pci_set = []
+        pci_target_set = []
+        for rmd in (rmd_u, rmd_b0, rmd_b90, rmd_b180, rmd_b270, rmd_p):
+            if rmd is not None:
+                pbp_set.append(
+                    find_one(
+                        "$.output.total_proposed_building_energy_cost_including_renewable_energy",
+                        rmd,
+                    )
+                )
+                bbp_set.append(
+                    find_one(
+                        "$.output.baseline_building_performance_energy_cost",
+                        rmd,
+                    )
+                )
+                pbp_nre_set.append(
+                    find_one(
+                        "$.output.total_proposed_building_energy_cost_excluding_renewable_energy",
+                        rmd,
+                    )
+                )
+                pci_set.append(
+                    find_one(
+                        "$.output.performance_cost_index",
+                        rmd,
+                    )
+                )
+                pci_target_set.append(
+                    find_one(
+                        "$.output.performance_cost_index_target",
+                        rmd,
+                    )
+                )
+
+        pbp_set = list(set(filter(lambda x: x is not None, pbp_set)))
+        bbp_set = list(set(filter(lambda x: x is not None, bbp_set)))
+        pbp_nre_set = list(set(filter(lambda x: x is not None, pbp_nre_set)))
+        pci_set = list(set(filter(lambda x: x is not None, pci_set)))
+        pci_target_set = list(set(filter(lambda x: x is not None, pci_target_set)))
+
+        assert_(len(pci_set) >= 1, "At least one `pci_set` value must exist.")
+        assert_(
+            len(pci_target_set) >= 1, "At least one `pci_target_set` value must exist."
+        )
+
+        return {
+            "pbp_set": pbp_set,
+            "bbp_set": bbp_set,
+            "pbp_nre_set": pbp_nre_set,
+            "pci_set": pci_set,
+            "pci_target_set": pci_target_set,
+        }
+
+    def rule_check(self, context, calc_vals=None, data=None):
+        pbp_set = calc_vals["pbp_set"]
+        bbp_set = calc_vals["bbp_set"]
+        pbp_nre_set = calc_vals["pbp_nre_set"]
+        pci_set = calc_vals["pci_set"]
+        pci_target_set = calc_vals["pci_target_set"]
+
+        return (
+            len(pbp_set)
+            == len(bbp_set)
+            == len(pbp_nre_set)
+            == len(pci_set)
+            == len(pci_target_set)
+            == 1
+            and bbp_set[0] != 0
```

Comment:
Not sure if you need to check this value or not, an assertion might be more appropriate than `FAIL`?

Reply:
Addressed.